### PR TITLE
Added journald configuration

### PR DESCRIPTION
**Pull request by luckyj5**
## Proposed changes
Added journald related config

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ x] I have read the [CONTRIBUTING](https://github.com/splunk/sck-otel/blob/develop/CONTRIBUTING.md) doc
- [x ] I have read the [CLA](https://github.com/splunk/sck-otel/blob/develop/CLA.md)
- [ x] I have added necessary documentation (if appropriate)
- [ x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: charts/sck-otel/templates/_pod.tpl [hunk 1]
```diff
@@ -94,6 +98,11 @@ containers:
         readOnly: true
       {{- end }}
 volumes:
+  {{- if .Values.journaldLogs.enabled }}
+  - name: journalpath
+    hostPath:
+      path: /run/log/journal
```
Comment (rockb1017) [inline charts/sck-otel/templates/_pod.tpl]: The path shouldn't be hardcoded. it should take `.Values.journaldLogs.directory`

File: charts/sck-otel/templates/_config.tpl [hunk 2]
```diff
@@ -84,6 +84,14 @@ extensions:
 #   The simplest way to specify the ballast size is set the value of SPLUNK_BALLAST_SIZE_MIB env variable.
     size_mib: ${SPLUNK_BALLAST_SIZE_MIB}
 receivers:
+
+# https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/journaldreceiver
+  {{- if .Values.journaldLogs.enabled }}
+  journald:
+    directory: {{- toYaml .Values.journaldLogs.directory | nindent 6 }}
+    units: {{- toYaml .Values.journaldLogs.units | nindent 8 }}
+    priority: {{- toYaml .Values.journaldLogs.priority | nindent 6 }}
```
Comment (rockb1017) [inline charts/sck-otel/templates/_config.tpl]: why do you need `nindent` ? 
`units` is an arrayy. does it get handled well ?

**Review by rockb1017 — CHANGES_REQUESTED** [review]
Looks good! let's fix some nits

File: charts/sck-otel/templates/_config.tpl (hunk 2 above)
Comment (luckyj5) [reply charts/sck-otel/templates/_config.tpl]: for indentation and new line to get the below format? 
```
receivers:
  journald:
    directory: /run/log/journal
    priority: info
    units:
    - kubelet
    - docker
    - containerd
```
Should this be handled using diff helm functions?

File: charts/sck-otel/templates/_pod.tpl (hunk 1 above)
Comment (luckyj5) [reply charts/sck-otel/templates/_pod.tpl]: Done

File: charts/sck-otel/templates/_config.tpl [hunk 3]
```diff
@@ -368,4 +376,22 @@ service:
...
     {{- end }}
+    {{- if .Values.journaldLogs.enabled }}
+    logs/journald:
+      receivers:
+        - journald
+      processors:
+        - batch
+        - resource/splunk
+      exporters:
+        {{- if .Values.splunkPlatform.endpoint }}
```
Comment (rockb1017) [inline charts/sck-otel/templates/_config.tpl]: could you rebase and use `{{- if eq (include "splunk-otel-collector.sendLogsToSplunk" .) "true" }}`

File: charts/sck-otel/templates/_config.tpl [hunk 4]
```diff
@@ -368,4 +376,22 @@ service:
...
+        - splunk_hec/platform
+        {{- end }}
+        {{- if .Values.splunkObservability.logsEnabled }}
+        {{- if or .Values.splunkObservability.ingestUrl .Values.splunkObservability.realm }}
```
Comment (rockb1017) [inline charts/sck-otel/templates/_config.tpl]: you can use this one here as well `{{- if eq (include "splunk-otel-collector.sendLogsToO11y" .) "true" }}`

**Review by rockb1017 — CHANGES_REQUESTED** [review]

File: charts/sck-otel/templates/_config-Agent.tpl [hunk 5]
```diff
@@ -24,6 +24,19 @@ extensions:
...
 receivers:
+# https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/journaldreceiver
+  {{- if .Values.journaldLogs.enabled }}
+  journald:
+    directory: {{- toYaml .Values.journaldLogs.directory | nindent 6 }}
+    units: {{- toYaml .Values.journaldLogs.units | nindent 8 }}
+    priority: {{- toYaml .Values.journaldLogs.priority | nindent 6 }}
+    resource:
+      com.splunk.source: {{.Values.journaldLogs.directory}}
+      com.splunk.sourcetype: 'EXPR("kube:"+$._SYSTEMD_UNIT)'
```
Comment (rockb1017) [inline charts/sck-otel/templates/_config-Agent.tpl]: it should be `$$` for splunk distro otel

File: charts/sck-otel/templates/_config-Agent.tpl [hunk 6]
```diff
@@ -24,6 +24,19 @@ extensions:
...
+      com.splunk.index: {{ .Values.journaldLogs.index}}
```
Comment (rockb1017) [inline charts/sck-otel/templates/_config-Agent.tpl]: let's edit it so that it would default to splunkPlatform.index when not provided

Comment (rockb1017) [reply charts/sck-otel/templates/_config-Agent.tpl]: also add these to `resource` as well. 
```
    resource:
      {{- if .Values.clusterName }}
      k8s.cluster.name: {{ .Values.clusterName }}
      {{- end }}
      {{- if .Values.environment }}
      deployment.environment: {{ .Values.environment }}
      {{- end }}
      {{- if .Values.customMetadata }}
      {{- toYaml .Values.customMetadata | nindent 6 }}
      {{- end }}
```

Comment (rockb1017) [reply charts/sck-otel/templates/_config-Agent.tpl]: let's also see if we can set `host.name` to `K8S_NODE_NAME` env var here.

File: charts/sck-otel/templates/_config-Agent.tpl [hunk 7]
```diff
@@ -531,4 +544,19 @@ service:
         - splunk_hec/platformMetrics
         {{- end }}
     {{- end }}
+    {{- if .Values.journaldLogs.enabled }}
+    logs/journald:
+      receivers:
+        - journald
+      processors:
+        - batch
+        - resource/splunk
```
Comment (rockb1017) [inline charts/sck-otel/templates/_config-Agent.tpl]: this `resource/splunk` processor doesn't fit well to journald logs. lets remove from journald processors.

File: charts/sck-otel/templates/_config-Agent.tpl (hunk 6 above)
Comment (rockb1017) [reply charts/sck-otel/templates/_config-Agent.tpl]: from your screenshot, host field is `unknown`

**Review by rockb1017 — CHANGES_REQUESTED** [review]

File: charts/sck-otel/templates/_config-Agent.tpl [hunk 8]
```diff
@@ -531,4 +544,19 @@ service:
         - splunk_hec/platformMetrics
         {{- end }}
     {{- end }}
+    {{- if .Values.journaldLogs.enabled }}
```
Comment (rockb1017) [inline charts/sck-otel/templates/_config-Agent.tpl:558]: lets also add one more condition 
`splunk-otel-collector.collectLog` to be true. it ensures at least one of endpoint is available for logs

**Review by rockb1017 — APPROVED** [review]
Looks good! Thanks for your work!

**Pull request merged**